docs/trouble-shooting/README.md

```diff
@@ -0,0 +1,342 @@
+and IPv6.
+
+If no traffic works a good place to start is to check that traffic
+really enters the system.
+
+```
+kubectl exec -it -n red meridio-load-balancer-5785674d6-4hqsh -c fe -- sh
+apk add tcpdump
+tcpdump -ni nsm-1
+```
+
+In this example we assume an [Alpine](https://www.alpinelinux.org/)
+container and add the `tcpdump` program. Then we use it to trace
+traffic on the external interface.
+
+
+### Bird
+
+The [bird](https://bird.network.cz/) routing suite runs in the `fe`
+container.
+
+
+### Nfqlb
+
+The [nfqlb](https://github.com/Nordix/nfqueue-loadbalancer/)
+loadbalancer runs in the `load-balancer` container. Please check the
+`nfqlb` documentation, it will help to understand the function and
+thus to troubleshoot.
+
+First check the "flows";
+
+```
+kubectl exec -n red meridio-load-balancer-5785674d6-6cqhm -c load-balancer -- nfqlb flow-list
+[{
+  "name": "tshm-stream1-flow1",
+  "priority": 0,
+  "protocols": [ "tcp" ],
+  "dests": [
+    "::ffff:10.0.0.1/128",
+    "1000::1:a00:1/128"
+  ],
+  "dports": [
+    "5000-5003"
+  ],
+  "matches_count": 540,
+  "user_ref": "tshm-stream1"
+}]
+```
+
+This should match your Meridio "flows".
+
+Next to check is the targets which should be all applications in your
+Meridio "stream";
+
+```
+kubectl exec -n red meridio-load-balancer-5785674d6-6cqhm -c load-balancer -- nfqlb show --shm=tshm-stream1
+Shm: tshm-stream1
+  Fw: own=0
+  Maglev: M=9973, N=100
+   Lookup: 57 57 57 57 57 57 66 89 57 57 66 57 89 57 66 57 57 89 66 89 57 66 66 57 89...
+   Active: 48(47) 58(57) 67(66) 90(89)
+```
+
+The `--shm` parameter is taken from the "user_ref" in the flow.
+
+This is an example from a system with 4 targets. They are all there
+which is a good sign. The `Active` array contains "firewall marks"
+(fwmark) used to route traffic to different target PODs;
+
+```
+# kubectl exec -n red meridio-load-balancer-5785674d6-6cqhm -c load-balancer -- ip rule
+0:      from all lookup local
+96:     from all fwmark 0x5a lookup 90
+97:     from all fwmark 0x43 lookup 67
+98:     from all fwmark 0x3a lookup 58
+99:     from all fwmark 0x30 lookup 48
+100:    from 10.0.0.1 lookup 4096
+32766:  from all lookup main
+32767:  from all lookup default
+# kubectl exec -n red meridio-load-balancer-5785674d6-6cqhm -c load-balancer -- ip route show table 90
+default via 172.16.1.8 dev load-balan-274a 
+```
+
+The target with fwmark 90 is routed to IPv4 address `172.16.1.8`. This
+address (and all other target addresses) must be reachable from the
+loadbalancer;
+
+```
+kubectl exec -n red meridio-load-balancer-5785674d6-6cqhm -c load-balancer -- ping -c1 -W1 172.16.1.8
+ancer -- ping -c1 -W1 172.16.1.8
+PING 172.16.1.8 (172.16.1.8) 56(84) bytes of data.
+64 bytes from 172.16.1.8: icmp_seq=1 ttl=64 time=0.859 ms
+
+--- 172.16.1.8 ping statistics ---
+1 packets transmitted, 1 received, 0% packet loss, time 0ms
+rtt min/avg/max/mdev = 0.859/0.859/0.859/0.000 ms
+```
+
+Traffic to the target goes via a "proxy".
+
+
+## Proxy
```

Comment:
The source routes have to be verified in the proxy. it should contains 1 rule per VIP with a table assigned. And in each table there should be nexthops with the IPs of the load-balancers

Reply:
Will add those